### PR TITLE
[ie/youtube:tab] Fix playlists tab extraction

### LINKED ISSUE #11524: [youtube:tab] Tab/playlist extraction intermittently yielding 0 items
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Switzerland

### Provide a description that is worded well enough to be understood

I'm using yt-dlp on a script of mine and I realised that one of my tests was flakey. In particular, listing playlists for a given YouTube channel would fail about 33% of the time, but nothing in my code seemed amiss.

Searching through the issues to see if I was doing something wrong or there was someone having similar trouble, I found https://github.com/yt-dlp/yt-dlp/issues/11511 (not related to my issue per-se), so I tried the exact command that was suggested to see if I could replicate my issues there, and indeed, I'm having the same trouble using the CLI command.

More than half the times, running the command `yt-dlp -vU --print "%(id)s|%(title)s|%(playlist_index)s" --flat-playlist "https://www.youtube.com/@LibraryoftheUntold/playlists"` works fine. But occasionally, it silently fails with no output or warning. It just doesn't print anything. When using the Python library directly, the issue manifests as an empty list in the `entries` key, whilst all other fields seem to be populated as usual.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

#### Expected Behaviour

> Most of the time, this is the output.

```shell
❯ yt-dlp -vU --print "%(id)s|%(title)s|%(playlist_index)s" --flat-playlist "https://www.youtube.com/@LibraryoftheUntold/playlists"

[debug] Command-line config: ['-vU', '--print', '%(id)s|%(title)s|%(playlist_index)s', '--flat-playlist', 'https://www.youtube.com/@LibraryoftheUntold/playlists']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.11.04 from yt-dlp/yt-dlp [197d0b03b] (pip)
[debug] Python 3.13.0 (CPython arm64 64bit) - macOS-15.1-arm64-arm-64bit-Mach-O (OpenSSL 3.4.0 22 Oct 2024)
[debug] exe versions: ffmpeg 7.1 (setts), ffprobe 7.1
[debug] Optional libraries: Cryptodome-3.21.0, brotli-1.1.0, certifi-2024.08.30, mutagen-1.47.0, requests-2.32.3, sqlite3-3.47.0, urllib3-2.2.3, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1838 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.11.04 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.11.04 from yt-dlp/yt-dlp)
[youtube:tab] Extracting URL: https://www.youtube.com/@LibraryoftheUntold/playlists
[youtube:tab] @LibraryoftheUntold/playlists: Downloading webpage
[debug] [youtube:tab] Selected tab: 'playlists' (playlists), Requested tab: 'playlists'
[download] Downloading playlist: Library of the Untold - Playlists
[youtube:tab] Playlist Library of the Untold - Playlists: Downloading 9 items of 9
[debug] The information of all playlist entries will be held in memory
[download] Downloading item 1 of 9
PL_39VJI5VnWsEY21lkFbMtTte1FecamGO|Paranormal / Strange Happenings|1
[download] Downloading item 2 of 9
PL_39VJI5VnWtOERqIiMX_Yt3_wEwJ8J-O|Podcasts with Incredible Minds|2
[download] Downloading item 3 of 9
PL_39VJI5VnWslKfsegFKYnslK3aUMwfRQ|Zen|3
[download] Downloading item 4 of 9
PL_39VJI5VnWsdRQcmUh3tNuyrdo0RbkbN|Alchemical Transmutation|4
[download] Downloading item 5 of 9
PL_39VJI5VnWvcHp6c1aO1fq3ir-1-lwMK|Gnostic Thought|5
[download] Downloading item 6 of 9
PL_39VJI5VnWvbkUkqZa38hPoJtQEclDAW|Unexplained / Great Mysteries|6
[download] Downloading item 7 of 9
PL_39VJI5VnWsT05y2pHAmjJsTh-E1qS30|Quick Summaries|7
[download] Downloading item 8 of 9
PL_39VJI5VnWvNy-mRKjkhq_hxTzaWa0RT|Book of Wonderful Suffering -by J. W. Phipps|8
[download] Downloading item 9 of 9
PL_39VJI5VnWuC9OnKk86lBa_IjRHQWqHw|ALL Documentaries|9
[download] Finished downloading playlist: Library of the Untold - Playlists
```

#### Occasional wrong behaviour

> Sometimes, this happens. I haven't found a clear pattern when it does.

```shell
❯ yt-dlp -vU --print "%(id)s|%(title)s|%(playlist_index)s" --flat-playlist "https://www.youtube.com/@LibraryoftheUntold/playlists"

[debug] Command-line config: ['-vU', '--print', '%(id)s|%(title)s|%(playlist_index)s', '--flat-playlist', 'https://www.youtube.com/@LibraryoftheUntold/playlists']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.11.04 from yt-dlp/yt-dlp [197d0b03b] (pip)
[debug] Python 3.13.0 (CPython arm64 64bit) - macOS-15.1-arm64-arm-64bit-Mach-O (OpenSSL 3.4.0 22 Oct 2024)
[debug] exe versions: ffmpeg 7.1 (setts), ffprobe 7.1
[debug] Optional libraries: Cryptodome-3.21.0, brotli-1.1.0, certifi-2024.08.30, mutagen-1.47.0, requests-2.32.3, sqlite3-3.47.0, urllib3-2.2.3, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1838 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.11.04 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.11.04 from yt-dlp/yt-dlp)
[youtube:tab] Extracting URL: https://www.youtube.com/@LibraryoftheUntold/playlists
[youtube:tab] @LibraryoftheUntold/playlists: Downloading webpage
[debug] [youtube:tab] Selected tab: 'playlists' (playlists), Requested tab: 'playlists'
[download] Downloading playlist: Library of the Untold - Playlists
[youtube:tab] Playlist Library of the Untold - Playlists: Downloading 0 items
[debug] The information of all playlist entries will be held in memory
[download] Finished downloading playlist: Library of the Untold - Playlists
```

### PR DESCRIPTION
**Pull request by seproDev**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Not as much pain as expected

Fixes #11524

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/youtube.py [hunk 1]
```diff
@@ -5084,10 +5089,35 @@ def _playlist_entries(self, video_list_renderer):
...
+            **traverse_obj(renderer, {
+                'title': ('metadata', 'lockupMetadataViewModel', 'title', 'content', {str}),
+            }),
+            thumbnails=self._extract_thumbnails(renderer, (
+                'contentImage', 'collectionThumbnailViewModel', 'primaryThumbnail', 'thumbnailViewModel', 'image'), final_key='sources'))
+
     def _rich_entries(self, rich_grid_renderer):
+        if lookup_view_model := traverse_obj(rich_grid_renderer, ('content', 'lockupViewModel', {dict})):
+            entry = self._extract_lookup_view_model(lookup_view_model)
+            if entry:
```
Comment (Grub4K) [inline yt_dlp/extractor/youtube.py:5115]:
```suggestion
            if entry := self._extract_lookup_view_model(lookup_view_model):
```

File: yt_dlp/extractor/youtube.py [hunk 2]
```diff
@@ -5084,10 +5089,35 @@ def _playlist_entries(self, video_list_renderer):
...
+        content_id = renderer.get('contentId')
+        if not content_id:
+            return
+        entry_url = None
+        if renderer.get('contentType') in ('LOCKUP_CONTENT_TYPE_PLAYLIST', 'LOCKUP_CONTENT_TYPE_PODCAST'):
+            entry_url = f'https://www.youtube.com/playlist?list={content_id}'
+        else:
+            self.report_warning(
+                f'Unsupported lookup view model content type "{renderer.get("contentType")}"{bug_reports_message()}', only_once=True)
+            return
```
Comment (Grub4K) [inline yt_dlp/extractor/youtube.py:5102]:
```suggestion
        if renderer.get('contentType') not in ('LOCKUP_CONTENT_TYPE_PLAYLIST', 'LOCKUP_CONTENT_TYPE_PODCAST'):
            self.report_warning(
                f'Unsupported lookup view model content type "{renderer.get("contentType")}"{bug_reports_message()}', only_once=True)
            return
        entry_url = f'https://www.youtube.com/playlist?list={content_id}'
```

File: yt_dlp/extractor/youtube.py [hunk 3]
```diff
@@ -5084,10 +5089,35 @@ def _playlist_entries(self, video_list_renderer):
...
+        return self.url_result(
+            entry_url,
+            ie=YoutubeTabIE, video_id=content_id,
+            **traverse_obj(renderer, {
+                'title': ('metadata', 'lockupMetadataViewModel', 'title', 'content', {str}),
+            }),
```
Comment (Grub4K) [inline yt_dlp/extractor/youtube.py:5108]: this formatting is kinda odd
```suggestion
            entry_url, ie=YoutubeTabIE, video_id=content_id, title=traverse_obj(renderer, (
                'metadata', 'lockupMetadataViewModel', 'title', 'content', {str})),
```

File: yt_dlp/extractor/youtube.py [hunk 4]
```diff
@@ -4986,6 +4986,11 @@ def _grid_entries(self, grid_renderer):
         for item in grid_renderer['items']:
             if not isinstance(item, dict):
                 continue
+            if lookup_view_model := item.get('lockupViewModel'):
+                entry = self._extract_lookup_view_model(lookup_view_model)
+                if entry:
```
Comment (seproDev) [inline yt_dlp/extractor/youtube.py:4991]:
```suggestion
                if entry := self._extract_lookup_view_model(lookup_view_model):
```

File: yt_dlp/extractor/youtube.py [hunk 5]
```diff
@@ -4986,6 +4986,10 @@ def _grid_entries(self, grid_renderer):
         for item in grid_renderer['items']:
             if not isinstance(item, dict):
                 continue
+            if lookup_view_model := traverse_obj(item, ('lockupViewModel', {dict})):
```
Comment (bashonly) [inline yt_dlp/extractor/youtube.py:4989]:
```suggestion
            if lockup_view_model := traverse_obj(item, ('lockupViewModel', {dict})):
```

File: yt_dlp/extractor/youtube.py [hunk 6]
```diff
@@ -4986,6 +4986,10 @@ def _grid_entries(self, grid_renderer):
...
+                if entry := self._extract_lookup_view_model(lookup_view_model):
```
Comment (bashonly) [inline yt_dlp/extractor/youtube.py:4990]:
```suggestion
                if entry := self._extract_lockup_view_model(lookup_view_model):
```

File: yt_dlp/extractor/youtube.py [hunk 7]
```diff
@@ -5084,10 +5088,29 @@ def _playlist_entries(self, video_list_renderer):
                 continue
             yield self._extract_video(renderer)
 
+    def _extract_lookup_view_model(self, view_model):
```
Comment (bashonly) [inline yt_dlp/extractor/youtube.py:5091]:
```suggestion
    def _extract_lockup_view_model(self, view_model):
```

File: yt_dlp/extractor/youtube.py [hunk 8]
```diff
@@ -5084,10 +5088,29 @@ def _playlist_entries(self, video_list_renderer):
...
+            return
+        return self.url_result(
+            f'https://www.youtube.com/playlist?list={content_id}', ie=YoutubeTabIE, video_id=content_id,
+            title=traverse_obj(view_model, (
+                'metadata', 'lockupMetadataViewModel', 'title', 'content', {str})),
+            thumbnails=self._extract_thumbnails(view_model, (
+                'contentImage', 'collectionThumbnailViewModel', 'primaryThumbnail', 'thumbnailViewModel', 'image'), final_key='sources'))
+
     def _rich_entries(self, rich_grid_renderer):
+        if lookup_view_model := traverse_obj(rich_grid_renderer, ('content', 'lockupViewModel', {dict})):
```
Comment (bashonly) [inline yt_dlp/extractor/youtube.py:5107]:
```suggestion
        if lockup_view_model := traverse_obj(rich_grid_renderer, ('content', 'lockupViewModel', {dict})):
```

File: yt_dlp/extractor/youtube.py [hunk 9]
```diff
@@ -5084,10 +5088,29 @@ def _playlist_entries(self, video_list_renderer):
...
+            if entry := self._extract_lookup_view_model(lookup_view_model):
```
Comment (bashonly) [inline yt_dlp/extractor/youtube.py:5108]:
```suggestion
            if entry := self._extract_lockup_view_model(lookup_view_model):
```

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/youtube.py [hunk 10]
```diff
@@ -5084,10 +5088,29 @@ def _playlist_entries(self, video_list_renderer):
                 continue
             yield self._extract_video(renderer)
 
+    def _extract_lookup_view_model(self, view_model):
+        content_id = view_model.get('contentId')
+        if not content_id:
+            return
+        if view_model.get('contentType') not in ('LOCKUP_CONTENT_TYPE_PLAYLIST', 'LOCKUP_CONTENT_TYPE_PODCAST'):
+            self.report_warning(
+                f'Unsupported lookup view model content type "{view_model.get("contentType")}"{bug_reports_message()}', only_once=True)
```
Comment (bashonly) [inline yt_dlp/extractor/youtube.py:5097]:
```suggestion
                f'Unsupported lockup view model content type "{view_model.get("contentType")}"{bug_reports_message()}', only_once=True)
```

Comment (bashonly) [inline yt_dlp/extractor/youtube.py:5097]: also i would use a var for contentType since we use it in the msg too

**Review by bashonly — APPROVED** [review]

**Review by Grub4K — APPROVED** [review]

**Review by coletdjnz — APPROVED** [review]
lgtm

**Pull request merged**